syntax = "proto3";

package count_wordula;

service MapReduceDriver {
    rpc GetTask(WorkerStatus) returns (Task) {}
}

message WorkerStatus {
    string worker_id = 1;
    int32 worker_status = 2;
    int32 task_status = 3;
}

message Task {
    string task_id = 1;
    int32 task_type = 2;
    string input_dir = 3;
    string input_filename = 4;
    string output_dir = 5;
    int32 N = 6;
    int32 M = 7;
    int32 map_id = 8;
    int32 reduce_id = 9;
}

message ExecutedTask {
    int32 task_id = 1;
    int32 task_type = 2;
    string output_filename = 3;
    int32 task_status = 4;
}